src/tests/multihost/alltests/test_inmemory_cache.py

```diff
@@ -0,0 +1,329 @@
+
+    @staticmethod
+    def test_bz1484376(multihost):
+        """
+        :title: Add a configuration option to SSSD to disable the memory cache bz1484376
+                Set value of option Memcache timeout to 35
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1484376
+        :id: b64237b4-96ef-11ed-adb1-845cf3eff344
+        :steps:
+            1. Add a configuration option to SSSD to disable the memory cache
+            2. Set memcache_timeout=0
+            3. Set value of option Memcache timeout to 35
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        client = multihost.client[0]
+        client.run_command("rm -rvf /var/lib/sss/mc/*")
+        client.run_command("systemctl restart sssd")
+        # MC directory is empty, result as expected.
+        assert client.run_command("ls -A /var/lib/sss/mc/").stdout_text == ''
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'memcache_timeout': '35'}, action='update')
+        tools.clear_sssd_cache()
+        clear_only_domain_log(multihost)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        date_start = int(client.run_command('date "+%s"').stdout_text.split()[0])
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(10)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(10)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(10)
+        elapsed_second = int(client.run_command('date "+%s"').stdout_text.split()[0]) - date_start
+        if elapsed_second < 35:
+            client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+            with pytest.raises(AssertionError):
+                find_logs(multihost,  "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(6)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+
+    @staticmethod
+    def test_time_out_300(multihost):
+        """
+        :title: Set default value of option Memcache timeout to 300
+        :id: a0b9ee92-96fd-11ed-bc4e-845cf3eff344
+        :steps:
+            1. Set default value of option Memcache timeout to 300
+            2. Validate the user using sss cache
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+        """
+        client = multihost.client[0]
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'memcache_timeout': ''}, action='delete')
```

Comment:
You are changing the config, backupsssdconf fixture should be used to restore the config.